aws/resource_aws_ec2_traffic_mirror_session.go

```diff
@@ -0,0 +1,206 @@
+		input.SessionNumber = aws.Int64(int64(n.(int)))
+	}
+
+	if d.HasChange("traffic_mirror_filter_id") {
+		n := d.Get("traffic_mirror_filter_id")
+		input.TrafficMirrorFilterId = aws.String(n.(string))
+	}
+
+	if d.HasChange("traffic_mirror_target_id") {
+		n := d.Get("traffic_mirror_target_id")
+		input.TrafficMirrorTargetId = aws.String(n.(string))
+	}
+
+	var removeFields []*string
+	if d.HasChange("description") {
+		n := d.Get("description")
+		if "" != n {
+			input.Description = aws.String(n.(string))
+		} else {
+			removeFields = append(removeFields, aws.String("description"))
+		}
+	}
+
+	if d.HasChange("packet_length") {
+		n := d.Get("packet_length")
+		if nil != n && n.(int) > 0 {
+			input.PacketLength = aws.Int64(int64(n.(int)))
+		} else {
+			removeFields = append(removeFields, aws.String("packet-length"))
+		}
+	}
+	log.Printf("[DEBUG] removeFields %v", removeFields)
+
+	if d.HasChange("virtual_network_id") {
+		n := d.Get("virtual_network_id")
+		log.Printf("[DEBUG] VNI has change %v", n)
+		if nil != n && n.(int) > 0 {
+			input.VirtualNetworkId = aws.Int64(int64(n.(int)))
+		} else {
+			removeFields = append(removeFields, aws.String("virtual-network-id"))
+		}
+	}
+
+	log.Printf("[DEBUG] removeFields %v", removeFields)
+	if len(removeFields) > 0 {
+		input.SetRemoveFields(removeFields)
+	}
+
+	_, err := conn.ModifyTrafficMirrorSession(input)
+	if nil != err {
+		return fmt.Errorf("Error updating traffic mirror session %v", err)
+	}
+
+	return resourceAwsEc2TrafficMirrorSessionRead(d, meta)
+}
+
+func resourceAwsEc2TrafficMirrorSessionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	sessionId := d.Id()
+	input := &ec2.DescribeTrafficMirrorSessionsInput{
+		TrafficMirrorSessionIds: []*string{
+			&sessionId,
+		},
+	}
+
+	out, err := conn.DescribeTrafficMirrorSessions(input)
+	if nil != err {
+		return fmt.Errorf("Error describing traffic mirror session %v", sessionId)
+	}
+
+	if 0 == len(out.TrafficMirrorSessions) {
+		log.Printf("[WARN] EC2 Traffic Mirror Session (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	session := out.TrafficMirrorSessions[0]
+	d.Set("network_interface_id", session.NetworkInterfaceId)
+	d.Set("session_number", session.SessionNumber)
+	d.Set("traffic_mirror_filter_id", session.TrafficMirrorFilterId)
+	d.Set("traffic_mirror_target_id", session.TrafficMirrorTargetId)
+	d.Set("description", session.Description)
+	d.Set("packet_length", session.PacketLength)
+	d.Set("virtual_network_id", *session.VirtualNetworkId)
+
+	return nil
+}
+
+func resourceAwsEc2TrafficMirrorSessionDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	sessionId := d.Id()
+	input := &ec2.DeleteTrafficMirrorSessionInput{
+		TrafficMirrorSessionId: &sessionId,
+	}
+
+	_, err := conn.DeleteTrafficMirrorSession(input)
+	if nil != err {
+		return fmt.Errorf("Error deleting traffic mirror session %v", sessionId)
```

Comment:
This is not returning the API/SDK error for operators and code maintainers to troubleshoot the issue:

```suggestion
		return fmt.Errorf("error deleting EC2 Traffic Mirror Session (%s): %w", sessionId, err)
```